acir/src/circuit/directives.rs

```diff
@@ -58,7 +58,10 @@ pub enum Directive {
         bits: Vec<Witness>, // control bits of the network which permutes the inputs into its sorted version
         sort_by: Vec<u32>, // specify primary index to sort by, then the secondary,... For instance, if tuple is 2 and sort_by is [1,0], then a=[(a0,b0),..] is sorted by bi and then ai.
     },
-    Log(LogInfo),
+    Log {
+        is_trace: bool, // This field states whether the log should be further manipulated or simply displayed to standard output
```

Comment:
Why do we have two separate process, and not just one which basically notifies when the 'log witness' are solved? Then the caller can decide to print to standard output or to further manipulate them. This logic should not be in ACIR.

Reply:
I am not suggesting to remove the flag, sorry for the mi-understanding, the location of my comment is confusing. This is because my first reaction was to ask for a clarification on the flag meaning. But at the end, I think it is better to not handle it in the solve, and just solve the witness.
With the flag (from the directive), the caller should be able to handle it correctly, shouldn't he?

Reply:
Yes that is the case. We only pass the flag along with the solved witness info so that the caller can determine how they should handle the solved log directive. The process in `solve_directives` for `Directive::Log` remains the same for both traces and printlns.

Reply:
So currently we only support logs of individual fields, arrays, or strings (formatting and handling of complex types can come in a separate PR). In the case of one witness we don't have to construct a list of hex elements. We can simply format the respective witness value and return that the log has been solved. In the case of multiple witnesses this means we have an array, where we have to fetch a list of hex elements, and format them into one finalized string output. Both cases will form a `LogOutputInfo::FinalizedOutput(String)`, which the caller of `solve` will then handle. In the case of a string this is formatted during acir gen and should already be a `LogOutputInfo::FinalizedOutput`. If `Directive::Log` already has finalized output we can just return the string stored in the enum.

Reply:
Thank you for the explanation. For me you are processing the logs here and you should not. The Log directive should just hold the parameters (for now just the flag), and the witness to log.
When the gate is solved, i.e when all the witness are known, you can send the parameters and witness values to the caller. That's it. How/where to print, which language to use, etc... all this should be handled by the caller.

Reply:
Got it, I see now. Ok I will refactor this